### PR TITLE
ci: use matrix instead downstreams

### PR DESCRIPTION
**Pull request by v1v**
## What does this pull request do?

- Revert downstream design to support the traditional matrix.
- Fix workspace permissions after running the docker-compose. https://github.com/elastic/apm-agent-ruby/pull/795/commits/608084d0d5b25320a6129bb6f1893ee606db3279
- Retry from a vanilla workspace, rather than reusing the same workspace where the tests failed. https://github.com/elastic/apm-agent-ruby/pull/795/commits/e2d53c1c1936d85142f34928431ba2060b115b8f

## Why is it important?

Downstream is not fully supported in the CI, for instance, if a build of a downstream failed then the object is not a RunWrapper but a FlowInterruption, therefore the population of the tests is not a trivial thing.

Benefits, aggregate test reporting as a GH PR comment, no more clicks on the Jenkins UI, ideally you won't need this anymore but the PR comment.

## Tasks
- [x] Converge coverage. See [here](https://apm-ci.elastic.co/job/apm-agent-ruby/job/apm-agent-ruby-mbp/job/PR-795/14/cobertura/)
- [x] Tests run as expected.

## Related issues

Requires https://github.com/elastic/apm-agent-ruby/issues/798

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -5,21 +5,17 @@ import co.elastic.matrix.*
...
-@Field def rubyDownstreamJobs = [:]
+@Field def rubyTasksGen
 
 pipeline {
   agent { label 'linux && immutable' }
   environment {
     REPO = 'apm-agent-ruby'
     BASE_DIR = "src/github.com/elastic/${env.REPO}"
-    PIPELINE_LOG_LEVEL = 'INFO'
+    PIPELINE_LOG_LEVEL = 'DEBUG'
```
Comment (v1v) [inline .ci/Jenkinsfile:18]:
```suggestion
    PIPELINE_LOG_LEVEL = 'INFO'
```

File: .ci/Jenkinsfile [hunk 2]
```diff
@@ -337,26 +303,97 @@ def runBenchmark(version){
...
+}
+
+def convergeCoverage() {
+  stage('Converge') {
+    deleteDir()
+    unstash('source')
+    dir("${BASE_DIR}"){
+      rubyTasksGen.dumpMatrix('-')?.each {
+        unstash(normalise("coverage-${it}"))
+      }
```
Comment (v1v) [inline .ci/Jenkinsfile:393]: :(

I see a corner case when decoupling the tests versus the master tests. As the last one will be the one used to converge the output.

Comment (v1v) [reply .ci/Jenkinsfile:397]: https://github.com/elastic/apm-agent-ruby/pull/795/commits/0f0f197d08504f172526b8b3e8f6fdcf5fe21fcf
